### PR TITLE
[jit] Fix flipped PackedSequence outputs in script

### LINKED ISSUE #32605: [jit] PackedSequence flips some outputs
In this repro, returning `packed` shows that `pack_padded_sequence` is different in script/eager since `unsorted_indices` is `None` in script and `[1, 0]` in eager. `pad_packed_sequence` has a similar problem.

* Only happens when `enforce_sorted=False`
```python
class ExperimentalLSTM(torch.nn.Module):
    def __init__(self, input_dim, hidden_dim):
        super().__init__()

    def forward(self, input):
        # type: (Tensor)
        packed = torch.nn.utils.rnn.pack_padded_sequence(
            input=input, lengths=torch.tensor([1, 2]), enforce_sorted=False
        )
        return packed
        # packed is
        # PackedSequence
        #     data
        #     batch_sizes
        #     sorted_indices
        #     unsorted_indices (this is the only different one, it's None in script)

        # output, lengths = torch.nn.utils.rnn.pad_packed_sequence(
        #     sequence=packed, total_length=2
        # )
        # # lengths is flipped, so is output
        # return output[0]

lstm = ExperimentalLSTM(input_dim=2, hidden_dim=2)
lstm = lstm.eval()

batch_size = 2

input = torch.arange(batch_size * 2 * 2).view(batch_size, 2, 2).float()
script_lstm = torch.jit.script(lstm)
script_out = script_lstm(input)
torch._C._jit_set_inline_everything_mode(False)
# print(script_lstm.graph)
print("script result\n{}".format(script_out))

print("\n")
eager_out = lstm(input)
print("eager result\n{}".format(eager_out))
```

cc @suo

### PR DESCRIPTION
**Pull request by driazati**
Stacked PRs
 * **#32955 - [jit] Fix flipped PackedSequence outputs in script**
 * #32953 - [jit] Support properties on `Device`

Fixes #32605

Differential Revision: [D20165514](https://our.internmc.facebook.com/intern/diff/20165514/)

### CODE REVIEW

File: torch/nn/utils/rnn.py [hunk 1]
```diff
@@ -61,37 +61,6 @@ class PackedSequence(PackedSequence_):
     #
     # See the note above in doc string (starting with ":attr:`data` can be on
     # arbitrary device...").
-
-    def __new__(cls, data, batch_sizes=None, sorted_indices=None, unsorted_indices=None):
```
Comment (eellison) [inline torch/nn/utils/rnn.py]: I see that you have edited construction of PackedSequence within our library, but doesn't removing this break BC for any  user code that uses this API?

Comment (eellison) [inline torch/nn/utils/rnn.py]: Doesn't removing this break BC for user code ?

**Review by eellison — APPROVED** [review]
I guess this is a fine short-term fix. But ignoring `__new__` style constructors means that this a silently-wrong semantics issue (& why we've had all these confusing LSTM bugs). When classes have `__new__` constructors we error. Can you file an issue for this ? 

cc @jamesr66a

**Comment by nairbv** [comment]
Test failures on master look relevant:

```
Feb 27 05:25:12 ERROR [0.027s]: test_script_pack_padded_sequence (test_jit.TestScript)
Feb 27 05:25:12 ----------------------------------------------------------------------
Feb 27 05:25:12 Traceback (most recent call last):
Feb 27 05:25:12   File "/var/lib/jenkins/workspace/test/test_jit.py", line 10873, in test_script_pack_padded_sequence
Feb 27 05:25:12     script_seq, script_lengths = scripted_pack_padded_seq(x, seq_lens)
Feb 27 05:25:12   File "/opt/conda/lib/python3.6/site-packages/torch/testing/_internal/common_utils.py", line 88, in prof_func_call
Feb 27 05:25:12     return prof_callable(func_call, *args, **kwargs)
Feb 27 05:25:12   File "/opt/conda/lib/python3.6/site-packages/torch/testing/_internal/common_utils.py", line 85, in prof_callable
Feb 27 05:25:12     return callable(*args, **kwargs)
Feb 27 05:25:12 RuntimeError: mutableType(from) && mutableType(to) INTERNAL ASSERT FAILED at /var/lib/jenkins/workspace/torch/csrc/jit/passes/alias_analysis.cpp:788, please report a bug to PyTorch. 
```

will unland.

**Pull request closed without merging**